### DISCUSSION TITLE
Installation & Running problem

### DISCUSSION THREAD

**Discussion by JUliusQin** [JacobBumgarner/VesselVio#68] [Q&A] [answered]
I am having problem running the app. I have a Win11 and this is what error I got

C:\Users\junlo\Desktop\VesselVio-main>py VesselVio.py
Traceback (most recent call last):
  File "C:\Users\junlo\Desktop\VesselVio-main\VesselVio.py", line 18, in <module>
    import pyvista as pv
ModuleNotFoundError: No module named 'pyvista'

C:\Users\junlo\Desktop\VesselVio-main>python VesselVio.py
Traceback (most recent call last):
  File "VesselVio.py", line 18, in <module>
    import pyvista as pv
  File "C:\Users\junlo\.pyenv\pyenv-win\versions\3.8.8\lib\site-packages\pyvista\__init__.py", line 12, in <module>
    from pyvista.plotting import *
  File "C:\Users\junlo\.pyenv\pyenv-win\versions\3.8.8\lib\site-packages\pyvista\plotting\__init__.py", line 4, in <module>
    from .charts import Chart2D, ChartMPL, ChartBox, ChartPie
  File "C:\Users\junlo\.pyenv\pyenv-win\versions\3.8.8\lib\site-packages\pyvista\plotting\charts.py", line 13, in <module>
    from pyvista import _vtk
  File "C:\Users\junlo\.pyenv\pyenv-win\versions\3.8.8\lib\site-packages\pyvista\_vtk.py", line 196, in <module>
    from vtkmodules.vtkFiltersExtraction import (
ImportError: cannot import name 'vtkExtractEdges' from 'vtkmodules.vtkFiltersExtraction' (C:\Users\junlo\.pyenv\pyenv-win\versions\3.8.8\lib\site-packages\vtkmodules\vtkFiltersExtraction.cp38-win_amd64.pyd)

Can anyone help me with this?

**Comment by JacobBumgarner** [comment]
What version of PyVista do you have installed?

**Reply by JUliusQin** [reply]
Thank you for your response. I am new to these, so I googled how to find the version of pyvista on windows. That is what I got. I basically followed the installation guide, except i made a little change on the requirement.txt, changing to "PyQt5==5.12.3", because there is no such version at this time.
Thank you for your patience.

> python -c "import pyvista; print(pyvista.__version__)"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Users\junlo\.pyenv\pyenv-win\versions\3.8.8\lib\site-packages\pyvista\__init__.py", line 12, in <module>
    from pyvista.plotting import *
  File "C:\Users\junlo\.pyenv\pyenv-win\versions\3.8.8\lib\site-packages\pyvista\plotting\__init__.py", line 4, in <module>
    from .charts import Chart2D, ChartMPL, ChartBox, ChartPie
  File "C:\Users\junlo\.pyenv\pyenv-win\versions\3.8.8\lib\site-packages\pyvista\plotting\charts.py", line 13, in <module>
    from pyvista import _vtk
  File "C:\Users\junlo\.pyenv\pyenv-win\versions\3.8.8\lib\site-packages\pyvista\_vtk.py", line 196, in <module>
    from vtkmodules.vtkFiltersExtraction import (
ImportError: cannot import name 'vtkExtractEdges' from 'vtkmodules.vtkFiltersExtraction' (C:\Users\junlo\.pyenv\pyenv-win\versions\3.8.8\lib\site-packages\vtkmodules\vtkFiltersExtraction.cp38-win_amd64.pyd)

**Comment by JacobBumgarner** [comment]
Ok:

Can you run `pip show pyvista pyvistaqt vtk` for me? Thanks

**Reply by JUliusQin** [reply]
C:\Users\junlo>pip show pyvisa pyvistaqt vtk
WARNING: Package(s) not found: pyvisa
Name: pyvistaqt
Version: 0.4.0
Summary: pyvista qt plotter
Home-page: https://github.com/pyvista/pyvistaqt
Author: PyVista Developers
Author-email: info@pyvista.org
License: MIT
Location: c:\users\junlo\.pyenv\pyenv-win\versions\3.8.8\lib\site-packages
Requires: pyvista, QtPy
Required-by:
---
Name: vtk
Version: 9.2.2
Summary: VTK is an open-source toolkit for 3D computer graphics, image processing, and visualization
Home-page: https://vtk.org
Author: VTK developers
Author-email:
License: BSD
Location: c:\users\junlo\.pyenv\pyenv-win\versions\3.8.8\lib\site-packages
Requires: matplotlib, wslink
Required-by: pyvista


It seems like pyvisa was not installed? What version should I install?

**Reply by JacobBumgarner** [reply]
Sorry, typo on my end. `pyvista`

**Reply by JUliusQin** [reply]
Thank you! This is what I got
C:\Users\junlo>pip show pyvista pyvistaqt vtk
Name: pyvista
Version: 0.33.2
Summary: Easier Pythonic interface to VTK
Home-page: https://github.com/pyvista/pyvista
Author: PyVista Developers
Author-email: info@pyvista.org
License: MIT
Location: c:\users\junlo\.pyenv\pyenv-win\versions\3.8.8\lib\site-packages
Requires: appdirs, imageio, numpy, pillow, scooby, vtk
Required-by: pyvistaqt
---
Name: pyvistaqt
Version: 0.4.0
Summary: pyvista qt plotter
Home-page: https://github.com/pyvista/pyvistaqt
Author: PyVista Developers
Author-email: info@pyvista.org
License: MIT
Location: c:\users\junlo\.pyenv\pyenv-win\versions\3.8.8\lib\site-packages
Requires: pyvista, QtPy
Required-by:
---
Name: vtk
Version: 9.2.2
Summary: VTK is an open-source toolkit for 3D computer graphics, image processing, and visualization
Home-page: https://vtk.org
Author: VTK developers
Author-email:
License: BSD
Location: c:\users\junlo\.pyenv\pyenv-win\versions\3.8.8\lib\site-packages
Requires: matplotlib, wslink
Required-by: pyvista

**Reply by JacobBumgarner** [reply, marked as answer]
Please try installing `vtk==9.0.1`

**Reply by JUliusQin** [reply]
Thank you! I am now able to run it. I also have an additional question. I have a hyperstacked nd2 file taken with Nikon confocal microscopy. How will I able to use VesselVio with nd2 files?

**Reply by JacobBumgarner** [reply]
Unfortunately it is not compatible with `nd2` files. You can this file to a NiFTI file (`.nii`) using FIJI to load it into VesselVio. Also, please remember that the vasculature needs to be segmented in order to use it with the app.

**Reply by JUliusQin** [reply]
Hi, I am sorry but I still having problem using the APP. I believe I successfully segmented the vasculatures. I tried both tiff and nii file formats, but I still get the error. Could you offer some suggestions? 

  C:\Users\junlo\Desktop\VesselVio-main>python VesselVio.py
  QWindowsWindow::setGeometry: Unable to set geometry 1190x695+90+0 (frame: 1190x695+90+0) on QWidgetWindow/"QStackedWidgetClassWindow" on "\\.\DISPLAY1". Resulting geometry: 1190x695+730+324 (frame: 1190x695+730+324) margins: 0, 0, 0, 0)
  Traceback (most recent call last):
    File "C:\Users\junlo\Desktop\VesselVio-main\library\qt_threading.py", line 449, in run
      volume, image_shape = ImProc.load_volume(volume_file)
  TypeError: cannot unpack non-iterable NoneType object

**Reply by JacobBumgarner** [reply]
Sorry for the delay. 

Do you have a sharable slice of this image that I could test locally? If the file is loadable in FIJI/other programs, you should be able to view it in VesselVio.
